client/src/fabric/FabricConnection.ts

```diff
@@ -235,6 +233,82 @@ export abstract class FabricConnection implements IFabricConnection {
 
     }
 
+    public async upgradeChaincode(name: string, version: string, channelName: string, fcn: string, args: Array<string>): Promise<any> {
+
+        const transactionId: Client.TransactionId = this.gateway.getClient().newTransactionID();
+        const upgradeRequest: Client.ChaincodeInstantiateUpgradeRequest = {
+            chaincodeId: name,
+            chaincodeVersion: version,
+            txId: transactionId,
+            fcn: fcn,
+            args: args
+        };
+
+        const network: Network = await this.gateway.getNetwork(channelName);
+        const channel: Client.Channel = network.getChannel();
+
+        const instantiatedChaincode: Array<any> = await this.getInstantiatedChaincode(channelName);
+
+        const foundChaincode: any = instantiatedChaincode.find((chaincode: any) => {
```

Comment:
can we make the finding chaincode a seperate function as you do the same thing in the instantiate function?